### PR TITLE
[typescript-angular] fix: when model property name is sanitized, use instead original property name within quotes in mustache template

### LINKED ISSUE #19352: [BUG] [typescript-angular] [modelPropertyNaming] when sanitized, model property name gets forced into snake_case
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When I have a model property that will be sanitized e.g.
```
        "properties": {
          "property.name": {
            ...
          }
        }
```
the output will be

```
export interface CustomerResponse { 
    property_name: string;
}
```

This is clearly unfriendly, since it's lying about the openAPI contract for that given model.
What I'd expect for such cases in typescript is this result:

```
export interface CustomerResponse { 
    'property.name': string;
}

```

##### openapi-generator version

7.7.0 (master)

##### OpenAPI declaration file content or url
[JSON spec](https://gist.github.com/davidediak/3a2a45768c3ecbeec04bd31ddf9b1bbf)

##### Generation Details

On master run
`./run-in-docker.sh mvn package `
`./run-in-docker.sh generate -i spec.json -g typescript-angular --minimal-update --global-property models`

you'll get a misleading interface, with a snake_cased property name.

##### Suggest a fix

I'll open a PR to fix this soon.

### PR DESCRIPTION
**Pull request by davidediak**
Fixes #19352 

Technical committee

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by davidediak** [comment]
@wing328 do you plan on merging this soon or there are any issues?

**Comment by LPCmedia** [comment]
+ 1 on this feature please. in our case some props have - in them and this is a blocker.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java [hunk 1]
```diff
@@ -243,6 +243,7 @@ public class CodegenProperty implements Cloneable, IJsonSchemaValidationProperti
     private boolean hasDiscriminatorWithNonEmptyMapping;
     private CodegenComposedSchemas composedSchemas = null;
     private boolean hasMultipleTypes = false;
+    private boolean hasSanitizedName = false;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java:246]: thanks for the PR,

can you please add a 1-liner explaining what this property means?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 2]
```diff
@@ -3884,6 +3884,7 @@ public CodegenProperty fromProperty(String name, Schema p, boolean required, boo
 
         property.name = toVarName(name);
         property.baseName = name;
+        property.setHasSanitizedName(!property.baseName.equals(property.name));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:3887]: FYI. @OpenAPITools/generator-core-team

**Comment by wing328** [comment]
can we add a test model with the property name in question to the test spec used by the typescript-angular petstore (fake) sample so that we know what the output looks like and the result works without issue?

**Comment by wing328** [comment]
when you've time, can you also please PM me via Slack?

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java (hunk 1 above)
Comment (davidediak) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java:247]: done

**Comment by wing328** [comment]
as discussed, I think this enhancement should only be applied to "original" modelPropertyNaming (when such option is enabled)

you tried the "original" modelPropertyNaming option before but it didn't work for you.

i'll investigate further over the weekend.

**Comment by wing328** [comment]
closed via https://github.com/OpenAPITools/openapi-generator/pull/19508

thanks for the PR

please pull the latest master to give it a try when you've time

**Pull request closed without merging**